src/index.ts

```diff
@@ -1,21 +1,159 @@
+import { IncomingMessage } from 'http'
+import {
+  getCookieFromDocument,
+  canUseDOM,
+  setCookie,
+  getCookieFromRequest,
+} from './utilities'
+
 class Internationalization<T extends string> {
-  supportedLanguages: T[]
+  private readonly cookieName: string
+
+  supportedLanguages: Array<T>
   fallbackLanguage: T
 
-  constructor(supportedLanguages: T[], fallbackLanguage: T) {
+  /**
+   * ClientInternationalization utility package for working with languages
+   *
+   * @param supportedLanguages { Array<string> } - List of all supported languages
+   * @param fallbackLanguage { string } - Fallback language if no language matches
+   * @param cookieName { string } - Optional: Set a preferred cookie name
+   */
+  constructor(
+    supportedLanguages: Array<T>,
+    fallbackLanguage: T,
+    cookieName?: string
+  ) {
     this.supportedLanguages = supportedLanguages
     this.fallbackLanguage = fallbackLanguage
+
+    this.cookieName = typeof cookieName === 'string' ? cookieName : 'language'
   }
 
+  /**
+   * Detect the browser language of the client
+   * If no language was found or if none of the
+   * supported language matches a fallback language will be returned
+   *
+   * @return {string} - Detected language
+   */
   detectBrowserLanguage(): T {
-    const browserLanguage =
-      (navigator.languages && navigator.languages[0]) || navigator.language
+    // Check if the DOM is presented.
+    // If the DOM isn't accessible the website may be rendered on a server
+    if (canUseDOM()) {
+      // Get browser language
+      const browserLanguage =
+        (navigator.languages && navigator.languages[0]) || navigator.language
+
+      return (
+        this.supportedLanguages.find((lang) => lang === browserLanguage) ||
+        this.fallbackLanguage
+      )
+    }
+
+    // Return fallback language if no DOM was found
+    return this.fallbackLanguage
+  }
+
+  /**
+   * Save a language to a cookie
+   * If the set language isn't in the list of supported languages
+   * the fallback language will be returned
+   *
+   * @param language { string } - Language to set
+   */
+  setLanguage(language: string): void {
+    if (canUseDOM()) {
+      const cookieLanguage =
+        this.supportedLanguages.find((lang) => lang === language) ||
+        this.fallbackLanguage
+
+      setCookie(this.cookieName, cookieLanguage)
+    }
+  }
+
+  /**
+   * Save a language to a cookie
+   * This method will not check if the language you want to set is supported
+   * Please note! If you have changed the cookie name, you must change it here as well
+   *
+   * @param language { string } - Language to set
+   * @param cookieName { string } - Optional: Cookie name
+   */
+  static setLanguage(language: string, cookieName?: string): void {
```

Comment:
Do we need this static function?

Reply:
I thought this method could be useful to quickly change the language without having to create a new instance

Reply:
I would leave it out for now and add it only if see a need for it.